tools/bcl-test-importer/BCLTestImporter/RegisterTypeGenerator.cs

```diff
@@ -4,13 +4,55 @@
+		// we will remove the dict and we will use the refection based method.
+		static Dictionary<string, (List<string> namespaces, string testAssembly, string testType)> cache = new Dictionary<string, (List<string> assemblies, string testAssembly, string testType)> {
+			{"SystemNumericsTests", (new List<string>{ "MonoTests.System.Numerics", }, "MONOTOUCH_System.Numerics_test.dll", "MonoTests.System.Numerics.BigIntegerTest")},
+			{"SystemRuntimeSerializationTests", (new List<string>{ "MonoTests", }, "MONOTOUCH_System.Runtime.Serialization_test.dll", "MonoTests.XmlComparer")},
+			{"SystemXmlLinqTests", (new List<string>{ "MonoTests.System.Xml", }, "MONOTOUCH_System.Xml.Linq_test.dll", "MonoTests.System.Xml.ExtensionsTest")},
+			{"MonoSecurityTests", (new List<string>{ "MonoTests.System.Security.Cryptography", }, "MONOTOUCH_Mono.Security_test.dll", "MonoTests.System.Security.Cryptography.SHA224ManagedTest")},
+			{"SystemComponentModelDataAnnotationTests", (new List<string>{ "MonoTests.System.ComponentModel.DataAnnotations", }, "MONOTOUCH_System.ComponentModel.DataAnnotations_test.dll", "MonoTests.System.ComponentModel.DataAnnotations.AssociationAttributeTest")},
+			{"SystemJsonTests", (new List<string>{ "MonoTests.System", }, "MONOTOUCH_System.Json_test.dll", "MonoTests.System.JsonValueTests")},
+			{"MonoDataTdsTests", (new List<string>{ "MonoTests.Mono.Data.Tds", }, "MONOTOUCH_Mono.Data.Tds_test.dll", "MonoTests.Mono.Data.Tds.TdsConnectionPoolTest")},
+			{"MonoCSharpTests", (new List<string>{ "MonoTests.Visit", }, "MONOTOUCH_Mono.CSharp_test.dll", "MonoTests.Visit.ASTVisitorTest")},
+			{"SystemJsonMicrosoftTests", (new List<string>{ "MonoTests.System", }, "MONOTOUCH_System.Json.Microsoft_test.dll", "MonoTests.System.JsonValueTests")},
+			{"MonoParallelTests", (new List<string>{ "MonoTests.Mono.Threading", }, "MONOTOUCH_Mono.Parallel_test.dll", "MonoTests.Mono.Threading.SnziTests")},
+			{"MonoTaskletsTests", (new List<string>{ "MonoTests.System", }, "MONOTOUCH_Mono.Tasklets_test.dll", "MonoTests.System.ContinuationsTest")},
+			{"SystemThreadingTasksDataflowTests", (new List<string>{ "MonoTests", }, "MONOTOUCH_System.Threading.Tasks.Dataflow_test.dll", "MonoTests.TestScheduler")},
+			{"SystemJsonXunit", (new List<string>{ "System.Json.Tests", }, "MONOTOUCH_System.Json_xunit-test.dll", "System.Json.Tests.JsonArrayTests")},
+			{"SystemNumericsXunit", (new List<string>{ "System.Numerics.Tests", }, "MONOTOUCH_System.Numerics_xunit-test.dll", "System.Numerics.Tests.GenericVectorTests")},
+			{"SystemLinqXunit", (new List<string>{ "Microsoft.Test.ModuleCore", }, "MONOTOUCH_System.Xml.Linq_xunit-test.dll", "Microsoft.Test.ModuleCore.LtmContext")},
+			{"SystemRuntimeCompilerServicesUnsafeXunit", (new List<string>{ "System.Runtime.CompilerServices", }, "MONOTOUCH_System.Runtime.CompilerServices.Unsafe_xunit-test.dll", "System.Runtime.CompilerServices.UnsafeTests")},
```

Comment:
it looks like `namespaces` don't need to be a `List<T>` - they are all one element
in fact it looks like its not needed (at all) since you have the fully qualified type name inside the tuple ?

Reply:
I'll double check if it is needed.

Reply:
It is not needed and was removed in the last push.